commit aab52a0cbf98848eee986ea9cd968a80695714a6 (refs/changes/82/4982/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-10-19T15:10:02-07:00 (12 months ago)
    
    TRITON-893 sdcadm check-health should ignore cainstsvc errors during transition to remove CA

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 7952262..a466f20 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -3872,6 +3872,13 @@ SdcAdm.prototype.checkHealth = function checkHealth(opts, cb) {
                     return false;
                 }
 
+                // CA (cloud analytics) is being removed from core Triton in
+                // TRITON-884. During that transition, the "cabase" and
+                // "cainstsvc" agents on a CN are ignored for health checking.
+                if (inst.service === 'cabase' || inst.service === 'cainstsvc') {
+                    return false;
+                }
+
                 return true;
             });
 
@@ -3907,9 +3914,8 @@ SdcAdm.prototype.checkHealth = function checkHealth(opts, cb) {
             return;
         }
 
-        // there are a couple agents which don't actually have SMF services,
-        // so skip them
-        if (inst.service.match(/(agents_core|cabase)$/)) {
+        // These agents don't have an SMF service to check.
+        if (inst.service === 'agents_core') {
             next();
             return;
         }
@@ -3992,9 +3998,8 @@ SdcAdm.prototype.checkHealth = function checkHealth(opts, cb) {
             return;
         }
 
-        // there are a couple agents which don't actually have SMF services,
-        // so skip them
-        if (inst.service.match(/(agents_core|cabase)$/)) {
+        // These agents don't have an SMF service to check.
+        if (inst.service === 'agents_core') {
             next();
             return;
         }
